src/Ztm.WebApi/ICallbackRepository.cs

```diff
@@ -0,0 +1,15 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Net;
+
+namespace Ztm.WebApi
+{
+    public interface ICallbackRepository
+    {
+        Task<Callback> AddAsync(IPAddress ip, DateTime registeredTime, Uri url, CancellationToken cancellationToken);
+        Task<Callback> SetStatusAsCompletedAsync(Guid id, CancellationToken cancellationToken);
```

Comment:
We should not return the effected callback due to some database provider may need to do additional query to be able to retrieve it. So just return `Task` should be better.